SwaprHQ/stackly-ui
# [STK-209][FEAT] - Enable select network without a connected wallet

## Fixes: [STK-209](https://linear.app/swaprhq/issue/STK-209/enable-select-network-without-connected-wallet)

# Description
* Moved some code out from the Stackbox and into SelectNetwork (concern separation)
* Created custom code to handle the Chain switch as app state 
* Updated the way we consume the chain switch where it's needed

# Visual evidence
![image](https://github.com/SwaprHQ/stackly-ui/assets/21271189/e3150862-71ff-4550-b829-d5078171035b)
![image](https://github.com/SwaprHQ/stackly-ui/assets/21271189/ca46fb0d-0e11-4d69-93a6-3e440bbee7f1)

# How to test the changes

1) Pull this branch
2) Run the project locally
3) Go to Stackly dApp
4) Connect your wallet
5) You should be able to change the selected chain either by connecting the wallet or with your wallet disconnected from the dApp. The Stackbox, Strategies, and TokenPicker should display relevant information either way.
